src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNINpHandle.cs

```diff
@@ -160,7 +161,8 @@ public override uint Receive(out SNIPacket packet, int timeout)
 
                     if (packet.Length == 0)
                     {
-                        return ReportErrorAndReleasePacket(packet, 0, SNICommon.ConnTerminatedError, string.Empty);
+                        var e = new Win32Exception();
+                        return ReportErrorAndReleasePacket(packet, (uint)e.NativeErrorCode, 0, e.Message);
```

Comment:
I also added native error checking for connection terminated errors in TCP handle, for consistency.